test/src/main/java/org/apache/accumulo/test/ComprehensiveIT.java

```diff
@@ -133,6 +151,58 @@ public static void teardown() {
     SharedMiniClusterBase.stopMiniCluster();
   }
 
+  @Test
+  public void testEventualScan() throws Exception {
+    Properties props = new Properties();
+    props.putAll(getClientProps());
+    props.put(ClientProperty.SCAN_SERVER_SELECTOR.getKey(),
+        ConfigurableScanServerSelector.class.getName());
+    props.put(ClientProperty.SCAN_SERVER_SELECTOR_OPTS_PREFIX.getKey() + "profiles",
+        "[{'isDefault':true,'timeToWaitForScanServers' : '10d','maxBusyTimeout':'5m','busyTimeoutMultiplier':4,'attemptPlans':[{\"servers\":\"100%\", \"busyTimeout\":\"3ms\"}]}]");
+
+    try (AccumuloClient client = Accumulo.newClient().from(props).build()) {
+
+      String table = getUniqueNames(1)[0];
+      client.tableOperations().create(table);
+
+      // Start Scan Server
+      getCluster().getClusterControl().start(ServerType.SCAN_SERVER);
+
+      // generate 100 rows of data
+      write(client, table, generateMutations(0, 100, tr -> true));
+
+      // need flush table so that eventual scan can see it.
+      getCluster().getClusterControl().stop(ServerType.SCAN_SERVER);
```

Comment:
I'm not sure why you are starting and stopping the scan server multiple times. You should be able to start it at the beginning of the test and shut it down at the end. Am I missing something?

Reply:
`Property.SSERV_CACHED_TABLET_METADATA_EXPIRATION` is being set to "5s" above, so it should be the case that after that expiration the ScanServer refreshes its view of the metadata for the tablets you are scanning over. Because you called `client.tableOperations().flush()` a new file was created on the filesystem and attached to that tablet. I think you just need to wait longer than 5s (maybe wait 10s) to pick up the new version of the tablet metadata. I would suggest trying that instead of stopping/starting the ScanServers.